bug/snapshot.go

```diff
@@ -62,3 +64,27 @@ func (snap *Snapshot) SearchTimelineItem(hash git.Hash) (TimelineItem, error) {
 
 	return nil, fmt.Errorf("timeline item not found")
 }
+
+// append the operation author to the actors list
+func (snap *Snapshot) addActor(actor identity.Interface) {
+	for _, a := range snap.Actors {
+		if actor.Id() == a.Id() {
+			return
+		}
+	}
+
+	snap.Actors = append(snap.Actors, actor)
+}
+
+// append the operation author to the participants and actors list
+func (snap *Snapshot) addParticipant(participant identity.Interface) {
+	for _, p := range snap.Participants {
+		if participant.Id() == p.Id() {
+			return
+		}
+	}
+
+	// a participant is an actor too
+	snap.Actors = append(snap.Actors, participant)
```

Comment:
Yes, a participant is also an actor. But what is the name of your function ? ;)

For readability, it's much better to have the calling code do both `addParticipant` and `addActor` instead of obfuscating what's really happening.

Reply:
Agreed !